crypto/Pos/Crypto/Orphans.hs

```diff
@@ -8,51 +8,149 @@ import           Universum
 
 import qualified Cardano.Crypto.Wallet as CC
 import qualified Cardano.Crypto.Wallet.Encrypted as CC
+import           Codec.CBOR.Decoding (Decoder)
+import qualified Codec.CBOR.Encoding as E
+import           Crypto.Error (CryptoFailable (..))
+import qualified Crypto.PubKey.Ed25519 as Ed25519
 import qualified Crypto.SCRAPE as Scrape
 import           Crypto.Scrypt (EncryptedPass (..))
-import qualified Crypto.Sign.Ed25519 as Ed25519
 import           Data.Aeson (FromJSON (..), ToJSON (..))
+import           Data.Aeson.Types (Parser)
+import qualified Data.ByteArray as BA
+import qualified Data.ByteString as BS
 import           Data.Hashable (Hashable)
-import           Data.SafeCopy (base, deriveSafeCopySimple)
+import           Data.SafeCopy (SafeCopy (..), base, contain,
+                     deriveSafeCopySimple, safeGet, safePut)
+import qualified Data.Text as T
 import           Serokell.Util.Base64 (JsonByteString (..))
 
 import           Pos.Binary.Class (Bi (..), decodeBinary, encodeBinary)
 
-instance Hashable Ed25519.PublicKey
-instance Hashable Ed25519.SecretKey
-instance Hashable Ed25519.Signature
+instance Hashable Ed25519.PublicKey where
+  hashWithSalt salt pk = hashWithSalt salt $ (BA.convert pk :: BS.ByteString)
+
+instance Hashable Ed25519.SecretKey where
+  hashWithSalt salt pk = hashWithSalt salt $ (BA.convert pk :: BS.ByteString)
+
+instance Hashable Ed25519.Signature where
+  hashWithSalt salt pk = hashWithSalt salt $ (BA.convert pk :: BS.ByteString)
+
+
+instance Ord Ed25519.PublicKey where
+  compare x1 x2 = compare (BA.convert x1 :: BS.ByteString) (BA.convert x2 :: BS.ByteString)
+
+instance Ord Ed25519.SecretKey where
+  compare x1 x2 = compare (BA.convert x1 :: BS.ByteString) (BA.convert x2 :: BS.ByteString)
+
+instance Ord Ed25519.Signature where
+  compare x1 x2 = compare (BA.convert x1 :: BS.ByteString) (BA.convert x2 :: BS.ByteString)
+
+
+instance SafeCopy BA.Bytes where
+  putCopy s =
+    let
+      toByteString :: BA.Bytes -> BS.ByteString
+      toByteString = BA.convert
```

Comment:
I see a lot of `BA.convert` with a type hint. It may be useful to write a separate top-level helper that does the conversion for you. Then a lot of this code can be simplified. `putyCopy` would be `putCopy = contain . safePut . toByteString`. `compare` could be `compare = comparing toByteString`. etc